### ISSUE TITLE
Longer config.h? How is this supposed to work for the Longer LK5 Pro when the config files don't even have Longer as an option 

### ISSUE THREAD

**Issue by eugenejanvrin** [Guizz27/Marlin---Longer-LKx-Pro#17]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by jgbrown54** [comment]
The following lines in the Configuration.h file are all that need to be changed to select the type of printer.

// @section machine
#define LGT_MAC //  For Alphawise and Longer U30 pro, LK4/LK5 pro
#define LK4_Pro //  for LK4pro, U30 Pro
//#define LK5_Pro //  for LK5pro
#define LKPro_BLTOUCH // for LK4Pro and LK5Pro

#define LKPro_FW_VERSION "GC 1.2"
#define LCD_HEIGHT 4 // number of message lines in the wait screen
//#define DEBUG_DGUSLCD
//#define FILAMENT_RUNOUT_SENSOR_DEBUG

// Choose the name from boards.h that matches your setup
#ifndef MOTHERBOARD
  #define MOTHERBOARD BOARD_LONGER3D_LKx_PRO //  For Alphawise and Longer U30 pro LK4 pro with LGT motherboard
#endif

**Comment by eugenejanvrin** [comment]
wow im an idiot didnt realize the default configuration.h in the zip file was the one i needed thank you very much

**Comment by eugenejanvrin** [comment]
i guess im doing it wrong because after uploading the firmware i cant move any steppers or preheat.. i also see no other menu options

**Comment by Guizz27** [comment]
Have you also updated the screen firmware?
You also have to make sure the sd card used to update the screen firmware is formatted according to the specs (Max 32GB, formatted FAT32 with 4096KB clusters).
Instructions at the bottom of this [page](https://github.com/Guizz27/Marlin---Longer-LKx-Pro)

**Comment by eugenejanvrin** [comment]
I have not updated the screen yet but only because I'm not getting any temp readouts and I cant manually move any axis'.

**Comment by Guizz27** [comment]
Are you trying from the screen or from usb like with gcode or software because it will not work with the original screen firmware, it is not compatible.

Also, the firmware requires homing the axis before they can be moved.

Temp readout should still work from a usb connection although I never tested it without a screen firmware update.

**Comment by eugenejanvrin** [comment]
I will test the screen firmware now

**Comment by eugenejanvrin** [comment]
Well unfortunately I cannot format any of my sd cards to 4096 bytes :(

**Comment by Guizz27** [comment]
Try to open a Command Prompt (on Windows hit the windows key, then type "cmd") and enter the following command. 
`FORMAT D: /FS:FAT32 /Q /A:4096`
 Replace D: with the drive letter assigned to your SD card.

**Comment by eugenejanvrin** [comment]
OK

**Comment by eugenejanvrin** [comment]
Invalid parameter.. all my sd card are 32 gb.. i may have a smaller one ill look

**Comment by eugenejanvrin** [comment]
i miss typed something on the first try 
this is the message i get



C:\Users\Eugene>FORMAT G: /FS:FAT32 /Q /A:4096
Insert new disk for drive G:
and press ENTER when ready...
The type of the file system is FAT32.
QuickFormatting 28.8 GB
The specified cluster size is too small for FAT32.
Format failed.

**Comment by eugenejanvrin** [comment]
Victory!.. I used win10 disk management tool to delete partition of my sd card the added a new one but only 8 gb the was able to format at 4096 bytes.. every thing works fine.. i love the new look and setting up the bl touch will be much easier.. 

thank you both for your help

**Comment by Guizz27** [comment]
You beat me. I was trying to write instructions on how to re-partition. Glad you figured it out and you could do it. Also glad that it now works!

**Comment by eugenejanvrin** [comment]
its still thinks its an LK4 with a 220x220x300 build though.. i think i can figure that out though

**Comment by Guizz27** [comment]
Make sure you comment out LK4 and uncomment LK5 in Configuration.h
//#define LK4_Pro //  for LK4pro, U30 Pro
#define LK5_Pro //  for LK5pro

**Comment by eugenejanvrin** [comment]
im not sure what im ding wrong but when i comment the lk4 setting out and uncomment the lk5 settings in the build fails.. i know im doing something simple wrong but at his point im too tired to think.. im going to get some sleep.. please please feel free to comment with some suggestions on what im doing wrong.. im fairly new to this

**Comment by jgbrown54** [comment]
Why don't you just download one of the pre-compiled version from here: https://github.com/Guizz27/Marlin---Longer-LKx-Pro/releases/tag/2.1-GC-1.2

**Comment by eugenejanvrin** [comment]
jgbrown54 thank you very much.. i simply did not know it existed.. im pretty new to this and teaching myself as i go.. that worked took all of 45 seconds thank you again

**Comment by eugenejanvrin** [comment]
[Guizz27] do you know anything about adding dual z steppers to the firmware

**Comment by Guizz27** [comment]
I'm sorry, I don't know exactly the steps to enable and calibrate dual Z steppers.  But let me know if you find the information and need help to recompile the firmware.

**Comment by eugenejanvrin** [comment]
OK I will, i need to figure out the "pinout" for the Z2_STEPPER_MOTOR

**Comment by Guizz27** [comment]
According to an old pinout diagram it seems the second Z is simply in parallel with the first one since there seems to be only one stepper for both Z. 

 
[LGT_KIT_V1.0_004 PIN.pdf](https://github.com/Guizz27/Marlin---Longer-LKx-Pro/files/10441822/LGT_KIT_V1.0_004.PIN.pdf)

**Comment by eugenejanvrin** [comment]
Ive been trying to find picture again i found it but it was blurry.. seems i should just be able to copy the pin readout

**Comment by eugenejanvrin** [comment]
not an easy task so far

**Comment by Guizz27** [comment]
for the second stepper motor, you just connect it to the second Z connector on the board. Nothing else to do.
I don't know the steps to calibrate both axis (make sure they are at the same height) when both stepper motors are in parallel (use the same stepper driver).

**Comment by eugenejanvrin** [comment]
i had done that but nothing was happening. turns out i needed to swap the two middle wires on the connector

**Comment by eugenejanvrin** [comment]
can i use UBL with this firmware

**Comment by Guizz27** [comment]
I never tested it. There are no controls on the screen for UBL but I think it could be controlled with gcode commands using a remote host.
You can turn it on in the code and test it if you want.

**Comment by bob291** [comment]
Does this have the firmware for the BL touch? I installed a CR Touch, and now I'm not sure what to do. I designed and printed a bracket and re-pinned the connections. But I don't know what to do about the firmware. I have updated the Motherboard firmware to 0.4.0-Marlin2.x and the screen to 0.3.1-DW.

**Comment by Guizz27** [comment]
Yes there is a version for BLTouch in the releases page but this is not the official Longer software. If you use this firmware, you also have to upgrade your screen firmware (also in the releases page). Be sure you have a screen that is compatible if you choose to use this one. this firmware has more features than Longer's official one. Make sure you have a DW type screen (in your about screen).

This is the release page: https://github.com/Guizz27/Marlin---Longer-LKx-Pro/releases/tag/2.1-GC-1.2

Longer also has a BLTouch version of their firmware but the link seems broken. You may have to contact them.

**Comment by macheen** [comment]
I have an incompatible screen the one with a j after it. is there any way i can reconfigure this to work for me or where can i get a DW screen?

**Comment by jgbrown54** [comment]
You can get a screen from China on aliexpress (https://www.aliexpress.us/item/3256804359976899.html?). Make sure you get the touch screen. Or you can run without the display with something like the Pixie (https://Pixie-Server.com) and move your control to a browser.

**Comment by eugenejanvrin** [comment]
@Guizz27 do you have the configuration file you use when you created the .hex file for the lk5 with bl touch

**Comment by Guizz27** [comment]
You can modify the configuration.h file, near the top you will see defines for LK4Pro and LK5Pro. Comment LK4Pro and uncomment LK5Pro. you will also see a define for BLTouch. Uncomment if you have a BLTouch or comment out the line if you don't. For example, for LK5Pro with BLTouch:

//#define LK4_Pro //  for LK4pro, U30 Pro
#define LK5_Pro //  for LK5pro
#define LKPro_BLTOUCH // for LK4Pro and LK5Pro
